app/src/main/java/com/ncnf/user/FriendsRepository.java

```diff
@@ -97,6 +99,23 @@ Friend request to another user (identified by its ID)
         return db.whereArrayContains(USERS_COLLECTION_KEY, FRIENDS_KEY, uuid, User.class);
     }
 
+    /*
+        Remove from friends
+     */
+    public CompletableFuture<Boolean> removeFriend(String my_uuid, String other_uuid) {
+        CompletableFuture<Boolean> u1 = db.removeArrayField(this.path + my_uuid, FRIENDS_KEY, other_uuid);
+        CompletableFuture<Boolean> u2 = db.removeArrayField(USERS_COLLECTION_KEY + other_uuid, FRIENDS_KEY, my_uuid);
+        return combine(u1, u2);
+    }
+
+
+    /*
+        Get list of users based on username
+     */
+    public CompletableFuture<List<User>> searchFriends(String username){
+        return db.withFieldLike(USERS_COLLECTION_KEY, FIRST_NAME_KEY, username, User.class); // TODO : change FIRSTNAMEKEY to USERNAME_KEY
```

Comment:
I'm assuming these changes from FIRST_NAME to USER_NAME require modification in the database?

Reply:
Yeah we do not have a way for users to set their username yet.